controllers/chat/moveToAnon.js

```diff
@@ -94,87 +113,84 @@ const moveToAnon = async (req, res) => {
-      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code = anon_user_info_emoji_code;
-      newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name = anon_user_info_emoji_name;
-
-      newStateMemberWithAnonInfo[req.userId].user.name = 'Anonymous ' + anon_user_info_emoji_name;
-      newStateMemberWithAnonInfo[req.userId].user.username =
-        'Anonymous ' + anon_user_info_emoji_name;
-
-      await ChatAnonUserInfo.create({
-        channel_id: newChannel.id,
-        my_anon_user_id: req.userId,
-        target_user_id: prevTargetUser.user_id,
-        anon_user_info_color_code: anon_user_info_color_code,
-        anon_user_info_color_name: anon_user_info_color_name,
-        anon_user_info_emoji_code: anon_user_info_emoji_code,
-        anon_user_info_emoji_name: anon_user_info_emoji_name
-      });
+    const createdChannel = await newChannel.create();
+    if (!is_anon_to_anon) {
+      newStateMemberWithAnonInfo = newChannel.state.members;
+      if (checkChatAnonUserInfo) {
+        newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code =
+          checkChatAnonUserInfo?.anon_user_info_color_code;
+        newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name =
+          checkChatAnonUserInfo?.anon_user_info_color_name;
+        newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code =
+          checkChatAnonUserInfo?.anon_user_info_emoji_code;
+        newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name =
+          checkChatAnonUserInfo?.anon_user_info_emoji_name;
+        newStateMemberWithAnonInfo[req.userId].user.name =
+          'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
+        newStateMemberWithAnonInfo[req.userId].user.username =
+          'Anonymous ' + checkChatAnonUserInfo?.anon_user_info_emoji_name;
+      } else {
+        newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code =
+          anon_user_info_color_code;
+        newStateMemberWithAnonInfo[req.userId].anon_user_info_color_name =
+          anon_user_info_color_name;
+        newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_code =
+          anon_user_info_emoji_code;
+        newStateMemberWithAnonInfo[req.userId].anon_user_info_emoji_name =
+          anon_user_info_emoji_name;
+        newStateMemberWithAnonInfo[req.userId].user.name = 'Anonymous ' + anon_user_info_emoji_name;
+        newStateMemberWithAnonInfo[req.userId].user.username =
+          'Anonymous ' + anon_user_info_emoji_name;
+        await ChatAnonUserInfo.create({
+          channel_id: newChannel.id,
+          my_anon_user_id: req.userId,
+          target_user_id: prevTargetUser.user_id,
+          anon_user_info_color_code: anon_user_info_color_code,
+          anon_user_info_color_name: anon_user_info_color_name,
+          anon_user_info_emoji_code: anon_user_info_emoji_code,
+          anon_user_info_emoji_name: anon_user_info_emoji_name
+        });
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this section, there are several instances where you're directly accessing properties on objects without checking if the object exists first. This can lead to `TypeError: Cannot read property 'property' of undefined` errors. It's a good practice to check if an object exists before trying to access its properties.

For example, in line 156, you're accessing `newChannel.create()`. If `newChannel` is null or undefined, this will throw an error. The same applies for lines 158, 159, 161-171, 173-183, and 185-191.

Consider adding checks for these objects before accessing their properties.

```diff
- const createdChannel = await newChannel.create();
+ const createdChannel = newChannel && await newChannel.create();

- newStateMemberWithAnonInfo = newChannel.state.members;
+ newStateMemberWithAnonInfo = newChannel && newChannel.state.members;

- newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code =
-   checkChatAnonUserInfo?.anon_user_info_color_code;
+ newStateMemberWithAnonInfo[req.userId].anon_user_info_color_code =
+   checkChatAnonUserInfo && checkChatAnonUserInfo.anon_user_info_color_code;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->